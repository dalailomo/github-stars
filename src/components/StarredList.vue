<template>
  <v-card flat v-if="loading" class="profile">
    <v-card-text><Loading /></v-card-text>
  </v-card>
  <v-list two-line v-else-if="items.length > 0">
    <template v-for="(item, index) in items">
      <router-link
        @click.native="selectItem(item)"
        :to="{ name: 'repositoryDetails', params: { username: username, repository: item.name, owner: item.owner.login } }"
        :key="item.id"
      >
        <v-list-tile avatar :key="item.owner.avatar_url" @click="">
          <v-list-tile-avatar>
            <img :src="item.owner.avatar_url">
          </v-list-tile-avatar>
          <v-list-tile-content>
            <v-list-tile-title v-html="item.name"></v-list-tile-title>
            <v-list-tile-sub-title v-html="item.full_name"></v-list-tile-sub-title>
          </v-list-tile-content>
        </v-list-tile>
        <v-divider></v-divider>
      </router-link>
    </template>
  </v-list>
  <v-card flat v-else>
    <v-card-text>No results :(</v-card-text>
  </v-card>
</template>

<script>
import Loading from './Loading'
import { mapState, mapActions } from 'vuex'

export default {
  components: {
    Loading,
  },

  computed: {
    ...mapState({
      items: state => state.starredList.items,
      loading: state => state.starredList.loading,
      item: state => state.starredList.currentItem,
      username: state => state.searchForm.username,
    }),
  },

  methods: {
    ...mapActions([
      'actionSetCurrent',
      'actionGetStarred',
    ]),

    selectItem(item) {
      this.actionSetCurrent(item)
    },

    fetchData() {
      if (this.$route.params.username) {
        this.actionGetStarred(this.$route.params.username).then(() => {
          if (!this.$route.params.repository) return

          this.items.forEach(iterationItem => {
            if (this.$route.params.repository === iterationItem.name) {
              this.selectItem(iterationItem)
            }
          })
        }).catch(e => {
          console.log(e)
        })
      }
    },
  },

  watch: {
    '$route'(to, from) {
      if (to.params.username !== from.params.username) {
        this.fetchData()
      }
    },
  },

  mounted() {
    this.fetchData()
  },
}
</script>

<style scoped lang="less">
a {
  color: initial;
  text-decoration: none;
}
</style>

<style>
.router-link-active > div {
  background: #d4f3ff;
}
</style>
