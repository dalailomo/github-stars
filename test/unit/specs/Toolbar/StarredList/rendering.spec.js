import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import StarredList from '@/components/Toolbar/StarredList'
import { getWrapper } from './../../../helpers'
import { emptyState, stateWithItems, mocks } from './mocks'

Vue.use(Vuex)
Vue.use(Vuetify)

describe('Starred list component rendering', () => {
  const actions = {
    actionGetStarred: jest.fn(),
    actionSetCurrent: jest.fn(),
  }

  it('Renders list with no results', () => {
    const wrapper = getWrapper(StarredList, actions, emptyState, mocks)

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Renders list with some results', () => {
    const wrapper = getWrapper(StarredList, actions, stateWithItems, mocks)

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
