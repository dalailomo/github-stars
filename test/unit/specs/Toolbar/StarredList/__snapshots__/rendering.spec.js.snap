// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starred list component rendering Renders list with no results 1`] = `
<div class="card card--flat" style="height: auto;">
  <div class="card__text">No results :(</div>
</div>
`;

exports[`Starred list component rendering Renders list with some results 1`] = `
<div class="list list--two-line">
  <router-link to="[object Object]">
    <div>
      <a class="list__tile list__tile--link list__tile--avatar">
        <div class="list__tile__avatar">
          <div class="avatar" style="height: 40px; width: 40px;">
            <img src="url">
          </div>
        </div>
        <div class="list__tile__content">
          <div class="list__tile__title">Foo</div>
          <div class="list__tile__sub-title">FooMan</div>
        </div>
      </a>
    </div>
    <hr class="divider">
  </router-link>
  <router-link to="[object Object]">
    <div>
      <a class="list__tile list__tile--link list__tile--avatar">
        <div class="list__tile__avatar">
          <div class="avatar" style="height: 40px; width: 40px;">
            <img src="url2">
          </div>
        </div>
        <div class="list__tile__content">
          <div class="list__tile__title">Foo2</div>
          <div class="list__tile__sub-title">FooMan2</div>
        </div>
      </a>
    </div>
    <hr class="divider">
  </router-link>
  <router-link to="[object Object]">
    <div>
      <a class="list__tile list__tile--link list__tile--avatar">
        <div class="list__tile__avatar">
          <div class="avatar" style="height: 40px; width: 40px;">
            <img src="url3">
          </div>
        </div>
        <div class="list__tile__content">
          <div class="list__tile__title">Foo3</div>
          <div class="list__tile__sub-title">FooMan3</div>
        </div>
      </a>
    </div>
    <hr class="divider">
  </router-link>
</div>
`;
